internal/datanode/segment.go

```diff
@@ -45,9 +46,10 @@ type Segment struct {
 	startPos *internalpb.MsgPosition // TODO readonly
 	endPos   *internalpb.MsgPosition
 
-	pkFilter *bloom.BloomFilter //  bloom filter of pk inside a segment
-	minPK    primaryKey         //	minimal pk value, shortcut for checking whether a pk is inside this segment
-	maxPK    primaryKey         //  maximal pk value, same above
+	statsChanged bool               //  segment stats changed
```

Comment:
this variable maintained in `Segment` looks bad

Reply:
Fixed